src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -203,6 +203,49 @@ public Response createDatasetRegistration(
     }
   }
 
+  @PUT
+  @Consumes({MediaType.MULTIPART_FORM_DATA})
+  @Produces({MediaType.APPLICATION_JSON})
+  @Path("/study/{studyId}")
+  @RolesAllowed({ADMIN, CHAIRPERSON, DATASUBMITTER})
+  /*
+   * This endpoint accepts a json instance of a dataset-registration-schema_v1.json schema.
+   * With that object, we can fully update the study/datasets from the provided values.
+   */
+  public Response updateDatasetRegistration(
+      @Auth AuthUser authUser,
+      FormDataMultiPart multipart,
+      @PathParam("studyId") Integer studyId,
+      @FormDataParam("dataset") String json) {
+    try {
+      datasetRegistrationService.findStudyById(studyId);
+      Set<ValidationMessage> errors = jsonSchemaUtil.validateSchema_v1(json);
+      if (!errors.isEmpty()) {
+        throw new BadRequestException(
+            "Invalid schema:\n"
+                + String.join("\n", errors.stream().map(ValidationMessage::getMessage).toList()));
+      }
+
+      DatasetRegistrationSchemaV1 registration = jsonSchemaUtil.deserializeDatasetRegistration(
+          json);
+      User user = userService.findUserByEmail(authUser.getEmail());
+
+      // key: field name (not file name), value: file body part
+      Map<String, FormDataBodyPart> files = extractFilesFromMultiPart(multipart);
+
+      // Update study from registration
```

Comment:
I think I need something here to validate that we're not trying to delete any consent groups in the incoming payload.